src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,81 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.DukeException;
+
+/**
+ * Type of Task which includes a end date and end time.
+ */
+public class Deadline extends Task {
+    /** Deadline date object of task */
+    protected LocalDate byDate;
+    /** Deadline time object of task */
+    protected LocalTime byTime;
+
+    /**
+     * Constructor to create a Deadline task.
+     *
+     * @param description describes the details of the Deadline.
+     * @param by the end date and time of the Deadline.
+     * @throws DukeException when date and time inputs are invalid, or when no date and time is inputted.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+        try {
+            this.byDate = formatDate(by);
+            this.byTime = formatTime(by);
+
+        } catch (DateTimeParseException | IndexOutOfBoundsException e) {
+            throw new DukeException("Please input date and time in correct format: YYYY/MM/DD HH:MM");
+        }
+    }
+
+    /**
+     * Overloaded constructor used when Tasks are being read from the local Storage of the user.
+     *
+     * @param description describes the details of the Deadline.
+     * @param isDone determines whether task has been completed or not.
+     * @param by the end date and time of the Deadline.
+     * @throws DukeException when date and time inputs are invalid, or when no date and time is inputted.
+     */
+    public Deadline(String description, boolean isDone, String by) throws DukeException {
+        super(description, isDone);
+        try {
+            this.byDate = formatDate(by);
+            this.byTime = formatTime(by);
+
+        } catch (DateTimeParseException | IndexOutOfBoundsException e) {
+            throw new DukeException("Please input date and time in correct format: YYYY/MM/DD HH:MM");
+        }
+    }
+
+    private LocalTime formatTime(String by) {
```

Comment:
a JavaDoc could be added here to describe the format that your Time will turn into